mgo.go

```diff
@@ -0,0 +1,66 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+const (
+	dbName             = "db"
+	monthlyInfoColName = "mi"
+)
+
+type collection interface {
+	ReplaceOne(ctx context.Context, filter interface{}, replacement interface{}, opts ...*options.ReplaceOptions) (*mongo.UpdateResult, error)
+}
+
+type Client struct {
+	mgoClient *mongo.Client
+	C         collection
+}
+
+func NewClient(url string) (*Client, error) {
+	client, err := mongo.NewClient(options.Client().ApplyURI(url))
+	if err != nil {
+		return nil, err
+	}
+	return &Client{mgoClient: client}, nil
+}
+
+func (c *Client) Connect() error {
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+	if err := c.mgoClient.Connect(ctx); err != nil {
+		return fmt.Errorf("error connection with mongo:%q", err)
+	}
+	c.C = c.mgoClient.Database(dbName).Collection(monthlyInfoColName)
+	return nil
+}
+
+func (c *Client) Disconnect() error {
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+	return c.mgoClient.Disconnect(ctx)
+}
+
+// Store processes and stores the crawling results.
+func (c *Client) Store(cr CrawlingResult) error {
+
+	agmi := AgencyMonthlyInfo{AgencyID: cr.AgencyID, Month: cr.Month, Year: cr.Year, Crawler: cr.Crawler, Employee: cr.Employees}
```

Comment:
Precisa checar se o campo  `c.col` é diferente de nil e dar um erro se não for. Lembre-se que essa API pode ser invocada antes do método connect ter sido chamada isso não deve ser permitido.